flow-preferences/src/main/java/com/fredporciuncula/flow/preferences/Preference.kt

```diff
@@ -1,28 +1,72 @@
 package com.fredporciuncula.flow.preferences
 
+import android.content.SharedPreferences
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.FlowCollector
 
+/**
+ * Wrapper that complements [SharedPreferences] api more powerful and handful to use coroutines and kotlin [Flow]
+ */
 interface Preference<T> {
 
+  /**
+   * Represents the key, that is used by [SharedPreferences] to store a preference
+   */
   val key: String
 
+  /**
+   * Represents a value, that will be returned, if the value by specific [key] will not be represented in [SharedPreferences] storage
+   */
   val defaultValue: T
 
+  /**
+   * Typical method to synchronously get a value of type [T] from [SharedPreferences]
+   */
   fun get(): T
 
+  /**
+   * Typical method to synchronously set a value of type [T] in [SharedPreferences]
+   */
   fun set(value: T)
 
+  /**
+   * Asynchronous that sets a value of type [T] from [SharedPreferences]
+   * @return true if the new values were successfully written
+   * to persistent storage
+   */
   suspend fun setAndCommit(value: T): Boolean
```

Comment:
💡 Even though this is a suspend (and thus async) function, it's actually responsible for a [**synchronous** write](https://developer.android.com/reference/android/content/SharedPreferences.Editor#apply()). So I would focus on that instead, since the `suspend` there is already documenting itself. My suggestion:

>Same as [set], but the update is immediately committed and written out to persistent storage synchronously.